core/dbt/config/renderer.py

```diff
@@ -145,7 +157,25 @@ def render_packages(self, packages: Dict[str, Any], packages_specified_path: str
             return package_renderer.render_data(packages)
 
     def render_selectors(self, selectors: Dict[str, Any]):
-        return self.render_data(selectors)
+        # Selectors require vars to be provided (unlike dbt_project.yml during deps)
+        # Create a strict context that will raise errors for missing vars
+        strict_ctx_obj: Union[TargetContext, BaseContext]
+        if self.profile:
+            strict_ctx_obj = TargetContext(
+                self.profile.to_target_dict(), self.cli_vars, require_vars=True
+            )
+        else:
+            strict_ctx_obj = BaseContext(self.cli_vars, require_vars=True)
+        strict_context = strict_ctx_obj.to_dict()
+
+        # Temporarily use strict context for selector rendering
+        original_context = self.context
+        self.context = strict_context
+        try:
+            return self.render_data(selectors)
+        finally:
+            # Restore lenient context
+            self.context = original_context
```

Comment:
I believe this change is no longer necessary 🤔 I actually went and tested it, and all the tests pass even if this change is reversed. I think this was only necessary when we were _always_ setting `require_vars=False` in the init. Now that we set `required_vars` based on the passed in `required_vars`, this temporarily being strict logic is no longer necessary